streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -678,6 +678,114 @@ trait ZSink[-R, +E, +A0, -A, +B] { self =>
         }
     }
 
+  final def zipPar[R1 <: R, E1 >: E, A2 >: A0, A1 <: A, C](
+    that: ZSink[R1, E1, A2, A1, C]
+  ): ZSink[R1, E1, A2, A1, (B, C)] =
+    new ZSink[R1, E1, A2, A1, (B, C)] {
+      type State = (Either[B, self.State], Either[C, that.State])
+
+      override def extract(state: State): ZIO[R1, E1, (B, C)] = {
+        val b: ZIO[R, E, B]   = state._1.fold(ZIO.succeed, s1 => self.extract(s1))
+        val c: ZIO[R1, E1, C] = state._2.fold(ZIO.succeed, s2 => that.extract(s2))
+        b.zip(c)
+      }
+
+      override def initial: ZIO[R1, E1, Step[State, Nothing]] =
+        self.initial.flatMap(s1 => {
```

Comment:
2 minor notes: 
- these types of blocks usually read nicer as `self.initial.flatMap { s1 =>`
- whenever you have multiple consecutive `flatMap`s, those usually read nicer as a `for` comprehension

Reply:
I do agree about using `for` and initially tried to write it like that. But I faced `no method withFilter` error and then continued with `flatMap`s